docs/resource/README.md

```diff
@@ -102,7 +102,104 @@ as specified in the [Resource SDK specification](https://github.com/open-telemet
+**[1]:** MUST be unique for each instance of the same `service.namespace,service.name` pair
+(in other words `service.namespace,service.name,service.instance.id` triplet MUST be globally unique).
+The ID helps to distinguish instances of the same service that exist at the same time
+(e.g. instances of a horizontally scaled service). It is preferable for the ID to be persistent
+and stay the same for the lifetime of the service instance, however it is acceptable that
+the ID is ephemeral and changes during important lifetime events for the service
+(e.g. service instance restarts).
+
+If the instance has no inherent unique ID that can be used as the value of this attribute,
+implementations MAY generate a random Version 1 or Version 4
+[RFC 4122](https://www.ietf.org/rfc/rfc4122.txt) UUID. When enough data is available,
+implementations SHOULD use Version 5 and SHOULD use the following UUID as the namespace:
+`4d63009a-8d0f-11ee-aad7-4c796ed8e320`.
+
+UUIDs are typically recommended, as only an opaque yet reproducible value for
+the purposes of identifying a service instance is needed. Similar to what can be seen in the man page for the
+[`/etc/machine-id`](https://www.freedesktop.org/software/systemd/man/machine-id.html) file,
+the underlying data, such as pod name and namespace should be treated as
+confidential by this algorithm, being the user's choice to expose it or not via another resource attribute.
+
+When a UUID v5 is generated, the input SHOULD be prefixed with
+`${telemetry.sdk.name}.${telemetry.sdk.language}.${service.namespace}.${service.name}`, followed by the
+the workload identifier, which should tentatively be stable. The workload identifier is the final piece of
+information making the source of telemetry to be uniquely idenfiable. For instance, it might be the `container.name`
+in containerized or Kubernetes environments. This means that the same service yields the
+same UUID if the same identifier (`host.id`, `/etc/machine-id`, and so on) remains the same. It would still
+yield different results for different services on the same host or namespace. When no namespaces or equivalent fields
+are available, the prefix MUST then be `${telemetry.sdk.name}.${telemetry.sdk.language}.${service.name}`.
+
+SDKs MUST provide a `service.instance.id`. SDKs SHOULD use the following algorithm when generating `service.instance.id`:
```

Comment:
> SDKs MUST provide a `service.instance.id`.

When the TC decided to split Semantic Conventions, we agreed to certain responsibilities of each specification.

TL;DR; This requirement MUST be moved into the OTEL specification.

Longer answer:

- The Semconv can define format + definition of attributes.
- The Specification defines behavior of SDK.

Since this would place a requirement that should be on the "spec-compliance-matrix" for SDKs, this needs to be defined in the OTEL Specification (likely under the `Resource` section).  I suggest we remove this sentence from semconv, and get your change merged.  We can then open a specification PR to require `service.isntance.id` and add the configuration component of it, where users can provide it via ENV variable or config file.

Reply:
This also affects the whole section of [Semantic Attributes with SDK-provided Default Value
](https://github.com/open-telemetry/semantic-conventions/tree/main/docs/resource#semantic-attributes-with-sdk-provided-default-value):

> ## Semantic Attributes with SDK-provided Default Value
>
> These are the attributes which MUST be provided by the SDK as specified in the [Resource SDK specification](https://github.com/open-telemetry/opentelemetry-specification/tree/v1.26.0/specification/resource/sdk.md#sdk-provided-resource-attributes):
>
> - [`service.name`](#service)
> - [`telemetry.sdk` group](#telemetry-sdk)

Reply:
@pyohannes, is there a ticket for this?

Reply:
Created: https://github.com/open-telemetry/opentelemetry-specification/issues/3839